### PR TITLE
fix_: usdt token decimals on tokenstore for arbitrum

### LINKED ISSUE #21738: Incorrect USDT estimation and decimals on swap on arbitrum network
### Precondition:
The user has no USDT balance on the Arbitrum network (this step is required)

### Steps to Reproduce:
1. Navigate to the Swap page.
2. Select Arbitrum as the network.
3. Set some asset "Assets to send" field value of which more than 1 USD (example: in my case 0.001 eth = 3 USD)
4. Observe the "Assets to receive" field and USDT decimals.

### Actual Result:
- Incorrect Estimation: USDT is estimated as <0.01 USD in the "Assets to receive" field
- Incorrect Decimals: USDT is displayed with 18 decimals instead of the correct 6 decimals, as specified in the [Arbiscan contract](https://arbiscan.io/token/0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9).

Image

### Expected Result:
- USDT is calculated correctly in the "Assets to receive" field, based on the value entered in "Assets to send."
- USDT decimals should display as 6 decimals, matching the token contract specification.

NOTE:
On desktop it works well
Image

### Devices:
Pixel 7a, Android 13
iPhone 11 Pro Max, iOS 17

### ENV:
Nightly Dec 2, 2024

### PR DESCRIPTION
**Pull request by briansztamfater**
This PR fixes a mismatch of token decimals for USDT on Arbitrum in token store. Should be 6 instead of 18.

From arbiscan:

Should fix [this status-mobile issue](https://github.com/status-im/status-mobile/issues/21738)

### CODE REVIEW

**Review by shivekkhurana — APPROVED** [review]

**Review by saledjenic — APPROVED** [review]
@briansztamfater could you update the same in `uniswap.go` for the chain id `56`, it's not in use atm, but anyhow, better to update it now, since we're going to add more chains very soon.

**Review by clauxx — APPROVED** [review]

**Review by clauxx — APPROVED** [review]

**Review by smohamedjavid — APPROVED** [review]

**Comment by briansztamfater** [comment]
> @briansztamfater could you update the same in `uniswap.go` for the chain id `56`, it's not in use atm, but anyhow, better to update it now, since we're going to add more chains very soon.

@saledjenic updated!

File: services/wallet/token/uniswap.go [hunk 1]
```diff
@@ -3799,7 +3799,7 @@ var uniswapTokens = []*Token{
 		Address:   common.HexToAddress("0x55d398326f99059fF775485246999027B3197955"),
 		Name:      "Tether USD",
 		Symbol:    "USDT",
-		Decimals:  18,
+		Decimals:  6,
 		ChainID:   56,
```
Comment (dlipicar) [inline services/wallet/token/uniswap.go]: this is for the Binance Smart Chain
https://bscscan.com/address/0x55d398326f99059ff775485246999027b3197955#readContract

18 is the right value here 
image

**Review by dlipicar — CHANGES_REQUESTED** [review]
see comment

File: services/wallet/token/uniswap.go (hunk 1 above)
Comment (dlipicar) [reply services/wallet/token/uniswap.go]: The entry in the uniswap token store for USDT on Arbitrum has the correct value already
image

In general, I'd say the Uniswap token list is as reliable as can be, we should never manually modify it.

Comment (briansztamfater) [reply services/wallet/token/uniswap.go]: True, just removed this change, only updating the Arbitrum one

**Review by dlipicar — APPROVED** [review]
I guess we should just remove some entries from our manually-maintained list to avoid errors like this and let Uniswap take care of keeping their list up to date. Anyway, we can do that work separately!

File: services/wallet/token/uniswap.go (hunk 1 above)
Comment (saledjenic) [reply services/wallet/token/uniswap.go]: Ohh I didn't check that. Thanks @dlipicar and sorry for the confusion @briansztamfater I just did the local search for USDT and saw all but that one with 6 decimals, my bad cause I haven't checked the contract.

Comment (briansztamfater) [reply services/wallet/token/uniswap.go]: No problem at all!

**Pull request merged**